CHANGELOG.md

```diff
@@ -28,6 +28,7 @@
       - `catalog_sync`
       - `storage_serialization_policy`
 - Added support for nested data types to `DataFrame.print_schema`
+- Added support for mixed case field names in struct type columns.
 
```

Comment:
does this only affected structured types or all `StructType`s? If it is the latter, then it would be a bcr, no?

Reply:
This only affects structured types because currently semi-structured objects don't become StructType columns, but instead get converted to MapType for some reason.